benchmarks/bench_pickle.py

```diff
@@ -0,0 +1,465 @@
+                      tries=args.tries)
+            del obj
+        if not args.nommap:
+            obj = [rnd.random_sample(a1_shape),
+                   generate_rand_dict(container_size,
+                                      with_arrays=args.valuearray,
+                                      with_string=args.valuestring,
+                                      array_shape=array_shape),
+                   rnd.random_sample(a2_shape)]
+            run_bench(bench_mmap, obj, name,
+                      mmap_mode=mmap_mode,
+                      tries=args.tries)
+            del obj
+
+if __name__ == "__main__":
+    import argparse
+    parser = argparse.ArgumentParser(description="Joblib benchmark script")
+    parser.add_argument('--compress', nargs='+', type=int, default=(0, 3),
+                        help="List of compress levels.")
+    parser.add_argument('--compressor', type=str, default='zlib',
+                        choices=['zlib', 'gzip', 'bz2', 'xz', 'lzma'],
+                        help="Compression algorithm.")
+    parser.add_argument('--mmap', type=str, default='r',
+                        choices=['r', 'r+', 'w+'],
+                        help="Memory map mode.")
+    parser.add_argument('--tries', type=int, default=5,
+                        help="Number of tries to compute execution time"
+                             "mean on.")
+    parser.add_argument('--shape', nargs='+', type=int, default=(10000, 10000),
+                        help="Big array shape.")
+    parser.add_argument("-m", "--nommap", action="store_true",
+                        help="Don't bench memmap")
+    parser.add_argument('--size', type=int, default=10000,
+                        help="Big dictionnary size.")
+    parser.add_argument('--valuearray', action="store_true",
+                        help="Use numpy arrays type in containers "
+                             "(list, dict)")
+    parser.add_argument('--valuearrayshape', nargs='+', type=int,
+                        default=(10, 10),
+                        help="Shape of arrays in big containers.")
+    parser.add_argument('--valuestring', action="store_true",
+                        help="Use string type in containers (list, dict).")
+    parser.add_argument("-n", "--nifti", action="store_true",
+                        help="Benchmark Nifti data")
+    parser.add_argument("-a", "--array", action="store_true",
+                        help="Benchmark single big numpy array")
+    parser.add_argument("-A", "--arrays", action="store_true",
+                        help="Benchmark list of big numpy arrays")
+    parser.add_argument("-d", "--dict", action="store_true",
+                        help="Benchmark big dictionnary.")
+    parser.add_argument("-l", "--list", action="store_true",
+                        help="Benchmark big list.")
+    parser.add_argument("-c", "--combo", action="store_true",
+                        help="Benchmark big dictionnary + list of "
```

Comment:
Very minor but this is dictionary with a single 'n'. You probably want do a global search and replace because you consistently wrote it with two 'n'. Noticed this when I was running the benchmarks.